reports/tests.py

```diff
@@ -528,3 +533,160 @@ def _generate_invoice_date(year=datetime.now().year, month=1, this_month=False):
+            sent_mail.to
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
+    def test_send_trial_instances_report(self):
+        """
+        Tests that if send_trial_instances_report succeeds, the recipient receives
+        a correct report with only instances created last month.
+        """
+        date1 = "2020-01-01"
+        date2 = "2020-02-01"
+        date3 = "2020-03-01"
+
+        with freeze_time(date1):
+            instance1 = self.create_user_with_trial_instance(key='1')
+        with freeze_time(date2):
+            instance2 = self.create_user_with_trial_instance(key='2')
+        with freeze_time(date3):
+            instance3 = self.create_user_with_trial_instance(key='3')
+
+            send_trial_instances_report(recipients=['to@example.com'])
+            sent_mail = mail.outbox[0]
+
+            self.assertEqual(len(mail.outbox), 1)
+
+            self.assertIn(
+                'Please find attached a CSV with the statistics for February 2020',
+                sent_mail.body
+            )
+
+            self.assertNotIn(
+                instance1.domain,
+                sent_mail.attachments[0][1]
+            )
+            self.assertIn(
+                instance2.domain,
+                sent_mail.attachments[0][1]
+            )
+            self.assertNotIn(
+                instance3.domain,
+                sent_mail.attachments[0][1]
+            )
+            self.assertEqual(
+                ['to@example.com'],
+                sent_mail.to
+            )
+
+    def test_send_trial_instances_report_corner_cases(self):
+        """
+        Tests that send_trial_instances_report succeeds handles date-based
+        corner cases well.
+        """
+        date1 = "2020-01-01 00:00:00"
+        date2 = "2020-01-31 23:59:59"
+        date3 = "2020-02-01 00:00:00"
+        date4 = "2020-02-29 23:59:59"
+        date5 = "2020-03-01 00:00:00"
+        date6 = "2020-03-31 23:59:59"
+
+        with freeze_time(date1):
+            instance1 = self.create_user_with_trial_instance(key='1')
+        with freeze_time(date2):
+            instance2 = self.create_user_with_trial_instance(key='2')
+        with freeze_time(date3):
+            instance3 = self.create_user_with_trial_instance(key='3')
+        with freeze_time(date4):
+            instance4 = self.create_user_with_trial_instance(key='4')
+        with freeze_time(date5):
+            instance5 = self.create_user_with_trial_instance(key='5')
+        with freeze_time(date6):
+            instance6 = self.create_user_with_trial_instance(key='6')
+
+            send_trial_instances_report(recipients=['to@example.com'])
+            sent_mail = mail.outbox[0]
+
+            self.assertEqual(len(mail.outbox), 1)
+
+            self.assertIn(
+                'Please find attached a CSV with the statistics for February 2020',
+                sent_mail.body
+            )
+            print(sent_mail.attachments[0][1])
```

Comment:
@spokerman12, nit: this should be removed.

Reply:
My bad!